File: Modules/HumanResource/OrganisationHierarchy/JobGrade/JobGradeForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.OrganisationHierarchy.Forms;

[FormScript("OrganisationHierarchy.JobGrade")]
[BasedOnRow(typeof(JobGradeRow), CheckNames = true)]
public class JobGradeForm
{
    public string Name { get; set; }
    public string Description { get; set; }
}

File: Modules/ServerTypes/OrganisationHierarchy/JobGradeForm.ts
import { StringEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface JobGradeForm {
    Name: StringEditor;
    Description: StringEditor;
}

export class JobGradeForm extends PrefixedContext {
    static formKey = 'OrganisationHierarchy.JobGrade';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!JobGradeForm.init)  {
            JobGradeForm.init = true;

            var w0 = StringEditor;

            initFormType(JobGradeForm, [
                'Name', w0,
                'Description', w0
            ]);
        }
    }
}